gems/smithy/lib/smithy/weld.rb

```diff
@@ -34,25 +35,34 @@ def post_process(artifacts)
     end
 
     # Called when constructing endpoint parameters. Any bindings defined here will
-    # be merged with other built-in bindings.
+    # be merged with other built-in bindings. The key is the name of the binding, and
+    # the value is the binding definition, which is a hash with keys :render_config,
+    # :render_build and :render_test_set.
     # @return [Hash<String, Hash>] endpoint built in bindings for use in endpoint rules.
     def endpoint_built_in_bindings
       {}
     end
 
     # Called when constructing the endpoint provider. Any bindings defined here will
-    # be merged with other function bindings.
+    # be merged with other function bindings. The key is the name of the binding, and
+    # the value is the function to call.
     # @return [Hash<String, String>] endpoint function bindings for use in endpoint rules.
     def endpoint_function_bindings
       {}
     end
 
     # Called when constructing the client. Any plugins defined here will be merged
-    # with other plugins.
-    # @return [Hash<String, String>] a mapping of fully qualified class names as the
-    #  key, and the plugin's file location as the value.
-    # def plugins
-    #   {}
-    # end
+    # with other plugins. The key is the fully qualified class name of the plugin,
+    # and the value is a hash with any of the following keys:
+    # * :source - the source code of the plugin
+    # * :path - the path to the plugin
```

Comment:
Whast the difference between path and require path?

Reply:
Path is where the plugin lives, and require path is how it's required in client.